sql/pre_install/tables.sql

```diff
@@ -53,6 +53,13 @@ CREATE TABLE _timescaledb_catalog.hypertable (
   compressed_hypertable_id integer,
   replication_factor smallint NULL,
   status int NOT NULL DEFAULT 0,
+  -- compression configuration
+  -- we use text[] instead of name[] as the latter will always pad every
+  -- element to NAMEDATALEN and therefore consume more space to store
+  compress_segmentby text[],
+  compress_orderby text[],
+  compress_orderby_desc bool[],
+  compress_orderby_nullsfirst bool[],
```

Comment:
To follow up:

Since a follow up PR will add per-chunk compression settings, I assume the plan is to put those settings in the `chunk` metadata table?

If that's the case, it would not be great for the same reasons as pointed out above.

Instead, I think there should be one normalized `compression_settings` table that has a pseudo-structure:

`<table_id, settings ...>`

Where `table_id` can be any table (chunk, hypertable root, or any other). Then you can simply put both "global", per-chunk, and regular table settings in the same table.

Reply:
The follow-up PR will get a dedicated table and the things currently stored in _timescaledb_internal.hypertable will serve as default for any subsequent compression operations. If there are no compression settings the fields are NULL which are stored in a bitmap in the header which consume no space.

Reply:
> By table_id you mean relid or did you have something else in mind?

Either relid or schema+name (which is our current way of representing tables).

Good to hear that the follow-up PR will store table settings in a dedicated table. Still, why separate default settings from other settings? If the default settings are the same as the hypertable "root" settings, then they would just as easily store fit in the dedicated compression settings table. It seems more complicated to spread compression settings across multiple tables  instead of having everything in one table, regardless of the settings being inherited (default) or not.

Reply:
Moved to separate table